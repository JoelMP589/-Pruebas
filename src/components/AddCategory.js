import React, { useState } from 'react'
import PropTypes from 'prop-types'

const AddCategory = ({ setCategorias }) => {
    const [texto, setTexto] = useState("")
    const handleSubmit = (e) => {
        e.preventDefault()
        if (texto.trim().length > 2) {
            setCategorias(categorias => [...categorias, texto])
            setTexto("")
        }
    }
    return (
        <form onSubmit={handleSubmit}>
            <input type="text" value={texto} onChange={(e) => { setTexto(e.target.value) }} placeholder="Categoria" />
        </form>
    )
}

AddCategory.propTypes = {
    setCategorias: PropTypes.func.isRequired
}

export default AddCategory