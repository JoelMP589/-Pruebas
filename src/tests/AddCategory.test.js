import { shallow } from "enzyme/build";
import AddCategory from "../components/AddCategory";

describe('pruebas en componente <AddCategory/>', () => {
    const setCategorias = jest.fn();
    const wrapper = shallow(<AddCategory setCategorias={setCategorias} />);
    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target: { value } });
        expect(wrapper.find('input').prop('value')).toBe(value);
    })
    test('no debe de agregar la categoria si el input esta vacio', () => {
        const input = wrapper.find('input');
        const value = '';
        input.simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategorias).not.toHaveBeenCalled();
    })
    test('debe de llamar el setCategorias y limpiar la caja de texto', () => {
        //hay que simular el evento onchange del input
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target: { value } });
        //haqye simular el evento onsubmit del form
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        //hay que esperar a que el setCategorias se ejecute
        expect(setCategorias).toHaveBeenCalled();
        expect(setCategorias).toHaveBeenCalledWith(expect.any(Function));
        expect(setCategorias).toHaveBeenCalledTimes(1);
        //verificvar que la caja de texto este vacia
        expect(wrapper.find('input').prop('value')).toBe('');
    })
})