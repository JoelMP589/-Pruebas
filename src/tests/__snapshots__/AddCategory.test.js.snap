// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en componente <AddCategory/> debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Categoria"
    type="text"
    value=""
  />
</form>
`;
